File: lockdemo/src/main/java/cn/iocoder/cloud/lockdemo/entity/BluetoothBase.java
package cn.iocoder.cloud.lockdemo.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 蓝牙设备信息实体类
 */
@Data
@TableName("bluetooth_base")
public class BluetoothBase {
    
    /**
     * 设备ID
     */
    @TableId(type = IdType.INPUT)
    private String id;
    
    /**
     * 锁具编号
     */
    private String lockNum;
    
    /**
     * MAC地址
     */
    private String mac;
    
    /**
     * 锁类型（0/2/3）
     */
    private String blueType;
    
    /**
     * 密码
     */
    private String password;
    
    /**
     * 密钥
     */
    private String secretKey;
    
    /**
     * 使用次数
     */
    private Integer useCount;
    
    /**
     * 上次使用人
     */
    private String lastUser;
    
    /**
     * 上次使用时间
     */
    private LocalDateTime lastUseTime;
    
    /**
     * 是否有关锁功能（0/1）
     */
    private String isInstructClosed;
    
    /**
     * 是否需要答题（0/1）
     */
    private String answerQuestion;
    
    /**
     * 电量
     */
    private Integer electQuantity;
    
    /**
     * 创建时间
     */
    private LocalDateTime createTime;
    
    /**
     * 更新时间
     */
    private LocalDateTime updateTime;
}

File: yudao-ui-admin-uniapp-master/src/api/lock/bluetooth-base.ts
import { http } from '@/http/http'

const baseUrl = '/digitalCard/bluetooth-base'

/** 蓝牙设备信息 */
export interface BluetoothBaseInfo {
  id: string
  lockNum: string
  mac: string
  blueType: string
  password: string
  secretKey: string
  useCount: number
  lastUser: string
  lastUseTime: string
  isInstructClosed: string
  answerQuestion: string
  electQuantity: number
}

/** 使用记录请求 */
export interface UseRecordReq {
  bluetoothId: string
  userId: string
  lockStatus: number // 1=开锁，0=关锁
}

/** 电量更新请求 */
export interface ElectQuantityReq {
  bluetoothId: string
  electQuantity: number
}

/** 获取设备信息 */
export function getBluetoothBaseInfo(id: string) {
  return http.get<BluetoothBaseInfo>(`${baseUrl}/getBluetoothBaseInfo`, { id }, undefined)
}

/** 添加使用记录 */
export function addUseRecords(data: UseRecordReq) {
  return http.post<void>(`${baseUrl}/addUseRecords`, data)
}

/** 更新电量 */
export function updateElectQuantity(data: ElectQuantityReq) {
  return http.post<void>(`${baseUrl}/updateElectQuantity`, data)
}

